IdentityCore/src/cache/mac/MSIDMacACLKeychainAccessor.m

```diff
@@ -70,26 +70,44 @@ - (nullable instancetype)initWithTrustedApplications:(nullable NSArray *)trusted
         {
             return nil;
         }
-
-        // Note: Apple seems to recommend serializing keychain API calls on macOS in this document:
-        // https://developer.apple.com/documentation/security/certificate_key_and_trust_services/working_with_concurrency?language=objc
-        // However, it's not entirely clear if this applies to all keychain APIs.
-        // Since our applications often perform a large number of cache reads on mulitple threads, it would be preferable to
-        // allow concurrent readers, even if writes are serialized. For this reason this is a concurrent queue, and the
-        // dispatch queue calls are used. We intend to clarify this behavior with Apple.
-        //
-        // To protect the underlying keychain API, a single queue is used even if multiple instances of this class are allocated.
-        static dispatch_once_t s_once;
-        dispatch_once(&s_once, ^{
-            s_synchronizationQueue = dispatch_queue_create("com.microsoft.msidmackeychaintokencache", DISPATCH_QUEUE_CONCURRENT);
-        });
-
+        
         MSID_LOG_WITH_CTX(MSIDLogLevelInfo, nil, @"Init MSIDMacACLPersistentTokenCache");
     }
 
     return self;
 }
 
++ (void)setSynchronizationQueue:(dispatch_queue_t)synchronizationQueue
```

Comment:
Does it mean MSAL CPP will need to call this before someone invokes synchronizationQueue getter?
Could it cause potential race conditions?

Reply:
OneAuth can call it anytime and ideally at startup where they can set the queue even before the class got init.

Since only one queue will be set during app session, it should be safe as long as the setter is called at the right place (before getter)

Reply:
Planned to call before init of the cache object. See this [commit](https://github.com/AzureAD/microsoft-authentication-library-for-cpp/commit/7178ad8373c6b054f2542976745dcd1c6bac9202) for reference. Should be good if no other place will call this getter function